### PR TITLE
[SPARK-38709][SQL] Remove trailing $ from function class name in sql-expression-schema.md

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

It's a bit weird to see class names like `CeilExpressionBuilder$` in `sql-expression-schema.md`. This PR removes the trailing `$`.

### Why are the changes needed?

code cleanup

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

existing tests

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @beliefer @MaxGekk

**Review by HyukjinKwon — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Review by wangyum — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Merged to master/3.3.

**Comment by dongjoon-hyun** [comment]
Oh.. After merging, I noticed that SPARK-38709 is filed as an improvement for 3.4.0.
Do you want me to revert it from branch-3.3, @cloud-fan ? Or, are we good?

**Comment by cloud-fan** [comment]
@dongjoon-hyun we are good. This is kind of test only change so OK to backport.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala [hunk 1]
```diff
@@ -184,10 +184,11 @@ object FunctionRegistryBase {
       } else {
         // This exists for the backward compatibility with old `ExpressionDescription`s defining
         // the extended description in `extended()`.
-        new ExpressionInfo(clazz.getCanonicalName, null, name, df.usage(), df.extended())
+        new ExpressionInfo(
+          clazz.getCanonicalName.stripSuffix("$"), null, name, df.usage(), df.extended())
```
Comment (beliefer) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala:188]: Could we extract `val canonicalName = clazz.getCanonicalName.stripSuffix("$")` ? It appears three times.

**Review by beliefer — COMMENTED** [review]
LGTM. Except one comment.

**Pull request closed without merging**